### ISSUE TITLE
SA: Access Violation with peds.ide

### ISSUE THREAD

**Issue by justalemon** [thelink2012/modloader#93]
I found this weird problem, where creating a peds.ide would crash the game with an Access Violation exception, as reported on modloader.log by the asi.

Just in case, I grabbed the stock peds.ide file from the stock game and wiped all of it's content and it still crashed.

The only other mod installed with the game is SilentPatch, everything else was disabled for testing this crash. Install was downgraded to 1.0 from Steam.

[modloader.log](https://github.com/thelink2012/modloader/files/6404962/modloader.log)
[peds.ide.zip](https://github.com/thelink2012/modloader/files/6404978/peds.ide.zip) (zipped because GitHub does not allows the upload of ide files)

**Comment by justalemon** [comment]
Looks like is replacing the entire peds.ide by the mod's peds.ide instead of using it to add new lines.

Is this intentional?

(Tested by copying the game's peds.ide and deleting the player line)

**Comment by thelink2012** [comment]
Yes, adding new lines are supposed to be done with readme lines. Using peds.ide behaves as if you replaced it in the data directory (with any necessary merges when multiple peds.ide exists)

**Comment by justalemon** [comment]
Ah, ok.

Is just that it wasn't mentioned in this repo nor the GTAForums post so I didn't knew.

It might be a good idea to document it, maybe on the GitHub Wiki?

**Comment by thelink2012** [comment]
There's some documentation already but I believe nobody ever finds them out. It comes bundled somewhere in the `modloader/.data` directory. [Example](https://github.com/thelink2012/modloader/blob/master/doc/plugins/gta3/std.data.md) for data loading stuff. To make those more discoverable and more informative is a good idea indeed.

**Comment by justalemon** [comment]
> but I believe nobody ever finds them out

Maybe because there is a bit of a mixed information across Markdown files? Let me explain a little bit:

* I didn't checked .data because I thought that it was only for modloader's configuration, maybe moving it to it's own directory like .docs would make it more visible and easier to find?
* I skipped doc/plugins/gta3/*.md as I thought that it was the documentation for modloader.h and/or gta3.hpp, which I was not looking for
* Reaching the documentation requires that you navigate across multiple directories instead of a single file or page, both here on the repo and the .data directory (doc/plugins/gta3/*.md for repo, modloader/.data/plugins/gta3/*.md for install)

> To make those more discoverable and more informative is a good idea indeed.

I have no problem with helping to move doc/plugins/gta3/*,md to the GitHub Wiki.

The md files of the wiki can then be added to the asi build as a submodule if required (https://github.com/thelink2012/modloader.wiki.git).

**Comment by thelink2012** [comment]
> Maybe because there is a bit of a mixed information across Markdown files? Let me explain a little bit:

Yeah. That wasn't meant to be offensive or anything, sorry if it sounded like that. I understand it's messy and don't expect people to find these files as easily as I thought it'd be back in the day.

> I have no problem with helping to move doc/plugins/gta3/*,md to the GitHub Wiki.

Sure. Go ahead =)

> The md files of the wiki can then be added to the asi build as a submodule if required (https://github.com/thelink2012/modloader.wiki.git).

No need. The Wiki can be linked from the bundled readme. The distribution should be less cluttered in the future by e.g. removing `modloader/.data` entirely. The plugin architecture will be replaced by a monolithic one and such.
